### PR TITLE
Retry juicefs umount

### PR DESCRIPTION
**Pull request by jsun-m**

### CODE REVIEW

File: pkg/storage/juicefs.go [hunk 1]
```diff
@@ -122,13 +123,29 @@ func (s *JuiceFsStorage) Format(fsName string) error {
...
 
 func (s *JuiceFsStorage) Unmount(localPath string) error {
-	cmd := exec.Command("juicefs", "umount", "--force", localPath)
+	juiceFsUmount := func() error {
+		cmd := exec.Command("juicefs", "umount", localPath)
 
-	output, err := cmd.CombinedOutput()
+		output, err := cmd.CombinedOutput()
+		if err != nil {
+			log.Printf("error executing juicefs umount: %v, output: %s", err, string(output))
```
Comment (luke-lombardi) [inline pkg/storage/juicefs.go:131]: Can you clean up this log

File: pkg/storage/juicefs.go [hunk 2]
```diff
@@ -122,13 +123,29 @@ func (s *JuiceFsStorage) Format(fsName string) error {
...
+		log.Printf("JuiceFS filesystem unmounted from: '%s'\n", localPath)
+		return nil
+	}
+
+	err := backoff.Retry(juiceFsUmount, backoff.WithMaxRetries(backoff.NewConstantBackOff(1*time.Second), 10))
+	if err == nil {
+		return nil
+	}
+
+	// Forcefully kill the fuse mount devices
```
Comment (luke-lombardi) [inline pkg/storage/juicefs.go:144]: Could this affect other fuse mounts of other workers running on multi-worker nodes?

**Review by luke-lombardi — CHANGES_REQUESTED** [review]

File: pkg/storage/juicefs.go (hunk 2 above)
Comment (jsun-m) [reply pkg/storage/juicefs.go:144]: It only kills pods that it finds using that file with fuser /dev/fuse which results to only files on that pod

File: pkg/storage/juicefs.go (hunk 1 above)
Comment (jsun-m) [reply pkg/storage/juicefs.go:131]: it's the same log as [here](https://github.com/beam-cloud/beta9/pull/402/files#diff-c6dd2c832240e75f3b4b734ef955e8cfeba0121667164d22f6bc10da2e881118L129)

**Review by luke-lombardi — APPROVED** [review]

**Pull request merged**